components/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/Utils.java

```diff
@@ -186,4 +192,29 @@ public static void checkSecurity() {
             secMan.checkPermission(new ManagementPermission("control"));
         }
     }
+
+    /**
+     * Returns a list of WSO2 Carbon Runtime names.
+     *
+     * @return WSO2 Carbon Runtime names
+     * @throws IOException if an I/O error occurs
+     */
+    public static List<String> getCarbonRuntimes() throws IOException {
+        Path carbonRuntimeHome = getCarbonHome().resolve(Constants.PROFILE_REPOSITORY);
+        Path osgiRepoPath = carbonRuntimeHome.resolve(Constants.OSGI_LIB);
+        if (!Files.exists(carbonRuntimeHome)) {
+            throw new IOException("invalid carbon home path " + carbonRuntimeHome.toString());
+        }
+        Stream<Path> runtimes = Files.list(carbonRuntimeHome);
+        List<String> runtimeNames = new ArrayList<>();
+
+        runtimes.parallel()
```

Comment:
parallel would not give much performance here as there will only be few runtimes (<5)